openslides_backend/services/datastore/adapter.py

```diff
@@ -278,23 +291,22 @@ def max(
             f"Start MAX request to datastore with the following data: {command.data}"
         )
         response = self.retrieve(command)
+        if lock_result:
+            self.update_locked_fields(
+                CollectionField(collection, field), response.get("position")
+            )
         return response
 
     def update_locked_fields(
         self,
-        key: Union[FullQualifiedId, FullQualifiedField],
+        key: Union[FullQualifiedId, FullQualifiedField, CollectionField],
         position: int,
     ) -> None:
         """
         Updates the locked_fields map by adding the new value for the given FQId or
-        FQField. If there is an existing value we take the smaller one.
+        FQField. To work properly in case of retry/reread we have to accept the new value always.
         """
-        current_position = self.locked_fields.get(str(key))
-        if current_position is None:
-            new_position = position
-        else:
-            new_position = min(position, current_position)
-        self.locked_fields[str(key)] = new_position
+        self.locked_fields[str(key)] = position
```

Comment:
Please check again: We have two cases: We always have to take the smallest position of a field to be locked if we are inside  an action (e. g. an action fetches the same model multiple times. The first position should be locked not the last.). If we are outside and want to retry in the retry-loop introduced with this PR of course the locked_fields property has to be reset.

Am I right?

Reply:
Yes, I'm really sorry, but I think you are right! If there are 2 values, for whatever reason, we have to take the smaller one. On retry we have to reset all locked_fields. I'll implement this in the continuing issue #131.

Reply:
Do you mean #351 ?

Reply:
yes, # 131 was the formerly number, before Finn reorganized the system.